mlrun/model_monitoring/stream_processing.py

```diff
@@ -0,0 +1,756 @@
+ERROR_COUNT = "error_count"
+ENTITIES = "entities"
+FEATURE_NAMES = "feature_names"
+LABEL_COLUMNS = "label_columns"
+LATENCY = "latency"
+RECORD_TYPE = "record_type"
+FEATURES = "features"
+PREDICTION = "prediction"
+PREDICTIONS = "predictions"
+NAMED_FEATURES = "named_features"
+NAMED_PREDICTIONS = "named_predictions"
+BASE_METRICS = "base_metrics"
+CUSTOM_METRICS = "custom_metrics"
+ENDPOINT_FEATURES = "endpoint_features"
+METRICS = "metrics"
+BATCH_TIMESTAMP = "batch_timestamp"
+TIME_FORMAT: str = "%Y-%m-%d %H:%M:%S.%f"  # ISO 8061
+
+
+# Stream processing code
+class EventStreamProcessor:
+    def __init__(
+        self,
+        project: str,
+        sample_window: int = 10,
+        tsdb_batching_max_events: int = 10,
+        tsdb_batching_timeout_secs: int = 60 * 5,  # Default 5 minutes
+        parquet_batching_max_events: int = 10_000,
+        parquet_batching_timeout_secs: int = 60 * 60,  # Default 1 hour
+        aggregate_count_windows: Optional[List[str]] = None,
+        aggregate_count_period: str = "30s",
+        aggregate_avg_windows: Optional[List[str]] = None,
+        aggregate_avg_period: str = "30s",
+        v3io_access_key: Optional[str] = None,
+        v3io_framesd: Optional[str] = None,
+        v3io_api: Optional[str] = None,
+    ):
+        self.project = project
+        self.sample_window = sample_window
+        self.tsdb_batching_max_events = tsdb_batching_max_events
+        self.tsdb_batching_timeout_secs = tsdb_batching_timeout_secs
+        self.parquet_batching_max_events = parquet_batching_max_events
+        self.parquet_batching_timeout_secs = parquet_batching_timeout_secs
+        self.aggregate_count_windows = aggregate_count_windows or ["5m", "1h"]
+        self.aggregate_count_period = aggregate_count_period
+        self.aggregate_avg_windows = aggregate_avg_windows or ["5m", "1h"]
+        self.aggregate_avg_period = aggregate_avg_period
+
+        self.v3io_framesd = v3io_framesd or config.v3io_framesd
+        self.v3io_api = v3io_api or config.v3io_api
+
+        self.v3io_access_key = v3io_access_key or environ.get("V3IO_ACCESS_KEY")
+        self.model_monitoring_access_key = (
+            os.environ.get("MODEL_MONITORING_ACCESS_KEY") or self.v3io_access_key
+        )
+
+        template = config.model_endpoint_monitoring.store_prefixes.default
+
+        kv_path = template.format(project=project, kind="endpoints")
+        _, self.kv_container, self.kv_path = parse_model_endpoint_store_prefix(kv_path)
+
+        tsdb_path = template.format(project=project, kind="events")
+        _, self.tsdb_container, self.tsdb_path = parse_model_endpoint_store_prefix(
+            tsdb_path
+        )
+        self.tsdb_path = f"{self.tsdb_container}/{self.tsdb_path}"
+
+        self.parquet_path = config.model_endpoint_monitoring.store_prefixes.user_space.format(
+            project=project, kind="parquet"
+        )
+
+        logger.info(
+            "V3IO Configuration",
```

Comment:
I suggest something more informative on where we are in the flow, e.g.
"Initializing event stream processor"